### PR TITLE
[WIP] Use FrameUI from framer.js

### PR DESCRIPTION
**Pull request by 3robi1**
Showing the FrameUI in the feed and user profile works properly with basic actions also working.
As I was not able to test locally signing, I left this as a WIP; will go over it again soon.

### CODE REVIEW

File: src/components/ui/frame.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            sizes="100vw"
+            height={0}
+            width={0}
+        />
+    );
+}
+
+
+export default function Frame( { homeFrameUrl, account } : { homeFrameUrl: string, account?: AccountObjectType }) {
+    const farcasterSigner: FarcasterSigner | undefined = account ? {
```
Comment (3robi1) [inline src/components/ui/frame.tsx:29]: @hellno I was not able to test this due do an endpoint error, under `/accounts` and __Connect__, which might be easier to sort out for you:

```json
{
    "code": "42501",
    "details": null,
    "hint": null,
    "message": "permission denied for function crypto_aead_det_encrypt"
}

File: src/components/ui/frame.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        fid: account?.id,
```
Comment (hellno) [inline src/components/ui/frame.tsx:30]: the fid for these herocast accounts is stored in `account.platformAccountId`.
to get the right type you can use 

```suggestion
        fid: Number(account?.platformAccountId),
```

File: src/components/ui/frame.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        status: account?.status,
+        publicKey: account?.publicKey,
+        privateKey: account?.privateKey,
+    } : undefined;
+
+    const frameState = useFrame({
+        homeframeUrl: homeFrameUrl,
+        connectedAddress: account?.publicKey,
```
Comment (hellno) [inline src/components/ui/frame.tsx:38]: connectedAddres should be picked from wagmi / viewm `useAccount`. The account.publicKey relates to the signer on the farcaster protocol https://docs.farcaster.xyz/learn/architecture/overview

File: src/components/ui/frame.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
+"use client";
```
Comment (hellno) [inline src/components/ui/frame.tsx:1]: can you move this file to `src > common > components`.
file structure isn't super clear, my bad. the `src > components > ui` only stores the shadcdn components right now.
will clarify this in the readme going forward

File: src/components/ui/frame.tsx (hunk 1 above)
Comment (hellno) [reply src/components/ui/frame.tsx:29]: I've seen this error before but have no immediate fix right now.
one step forward might be to run this sql script in the supabase local instance at `https://localhost:54323/`

```sql
GRANT
EXECUTE
  ON FUNCTION pgsodium.crypto_aead_det_encrypt (bytea, bytea, bytea, bytea) TO authenticated;

GRANT
EXECUTE
  ON FUNCTION pgsodium.crypto_aead_det_encrypt (bytea, bytea, uuid, bytea) TO authenticated;

GRANT
EXECUTE
  ON FUNCTION pgsodium.crypto_aead_det_decrypt (bytea, bytea, bytea, bytea) TO authenticated;

GRANT
EXECUTE
  ON FUNCTION pgsodium.crypto_aead_det_decrypt (bytea, bytea, uuid, bytea) TO authenticated;

GRANT
  pgsodium_keyiduser TO authenticated;
```

**Review by hellno — CHANGES_REQUESTED** [review]
great push on this. few minor changes so we can merge this 🙌🏼

File: src/components/ui/frame.tsx (hunk 3 above)
Comment (3robi1) [reply src/components/ui/frame.tsx:38]: oh yeah, thanks, will read Farcaster docs soon enough!

**Review by hellno — APPROVED** [review]
LGTM 🙌🏼

**Pull request merged**